src/phase0/state_transition/mod.rs

```diff
@@ -82,6 +82,28 @@ pub enum InvalidOperation {
     IndexedAttestation(InvalidIndexedAttestation),
     #[error("invalid deposit: {0}")]
     Deposit(InvalidDeposit),
+    #[error("invalid block header: {0}")]
+    HeaderValidation(InvalidBeaconBlockHeader),
+}
+
+#[derive(Debug, Error)]
+pub enum InvalidBeaconBlockHeader {
+    #[error("mismatch between state slot {state_slot} and block slot {block_slot}")]
+    StateSlotMismatch { state_slot: Slot, block_slot: Slot },
+    #[error("mismatch between the block's parent root {expected} and the expected parent root {provided}")]
+    ParentBlockRootMismatch { expected: Root, provided: Root },
```

Comment:
It doesn't compile since `ssz_rs::Node` does not satisfy `std::fmt::Display`. Should I implement it or it is better not to touch the `ssz_rs` code?

Reply:
ah, let's just use `Debug` here

```suggestion
    #[error("mismatch between the block's parent root {expected:?} and the expected parent root {provided:?}")]
    ParentBlockRootMismatch { expected: Root, provided: Root },
```

Reply:
Done: 5b790a5